update_list.py

```diff
@@ -0,0 +1,351 @@
+import argparse
+import json
+import re
+from enum import Enum
+from itertools import cycle
+from pathlib import Path
+from typing import Union
+
+import requests
+from user_agents.parsers import UserAgent, parse
+
+
+class Browser(Enum):
+    CHROME = "chrome"
+    FIREFOX = "firefox"
+    EDGE = "edge"
+    SAFARI = "safari"
+
+    @classmethod
+    def _missing_(cls, value):
+        """This method makes the Enum case-insensitive."""
+        value = str(value).lower()
+        for member in cls:
+            if member.value == value:
+                return member
+        return None
+
+
+class UserAgentUpdater:
+    UPDATE_USERAGENT_URL = r"https://user-agents.net/download"
+    USERAGENT_VERSION_URL = r"https://www.browsers.fyi/api/"
+
+    def __init__(
+        self,
+        output_folder: Union[str, Path],
+        timeout: float = 100.0,
+        output_name: str = "browsers.json",
+        append: bool = True,
+    ):
+        self.timeout = timeout
+        self.ua_list = []
+        self.output_name = output_name
+        self.append = append
+
+        if isinstance(output_folder, str):
+            output_folder = Path(output_folder)
+        if not output_folder.exists():
+            raise NameError("The specified folder does not exist.")
+        if not output_folder.is_dir():
+            raise TypeError("The specified path should be a folder.")
+        self.output_folder = output_folder
+
+        # Regex patterns
+        self.chrome_pattern = re.compile(r"chrome", re.IGNORECASE)
+        self.firefox_pattern = re.compile(r"firefox", re.IGNORECASE)
+        self.safari_pattern = re.compile(r"safari", re.IGNORECASE)
+        self.edge_pattern = re.compile(r"edge", re.IGNORECASE)
+
+        self.ios_pattern = re.compile(r"ios", re.IGNORECASE)
+        self.linux_pattern = re.compile(
+            r"Linux|Ubuntu|Arch|Fedora|OpenSuse|Debian", re.IGNORECASE
+        )
+        self.macos_pattern = re.compile(r"Mac", re.IGNORECASE)
+        self.windows_pattern = re.compile(r"Windows|win10|win11|win7", re.IGNORECASE)
+        self.android_pattern = re.compile(r"android", re.IGNORECASE)
+
+    def send_get_request(self, url: str):
+        response = requests.get(url, timeout=self.timeout)
+        response.raise_for_status()
+        result = response.text
+        return json.loads(result)
```

Comment:
You can use `return response.json()` directly, pretty sure.

Reply:
Fixed.